circuit-test/src/commonMain/kotlin/com/slack/circuit/test/CircuitReceiveTurbine.kt

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2024 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuit.test
+
+import app.cash.turbine.Event
+import app.cash.turbine.ReceiveTurbine
+import com.slack.circuit.runtime.CircuitUiState
+
+/**
+ * A Circuit-specific extension to [ReceiveTurbine] with extra helper functions for Circuit testing.
+ */
+public interface CircuitReceiveTurbine<UiState : CircuitUiState> : ReceiveTurbine<UiState> {
+  /** Awaits the _next_ [UiState] emission that is different than the [lastState]. */
+  public suspend fun consumeState(lastState: UiState? = tryMostRecentItem()): UiState
+}
+
+/**
+ * Awaits a subtype of [UiState] of type [SubState]. This is useful for sealed state type
+ * hierarchies.
+ *
+ * ```kotlin
+ * presenter.test {
+ *   val loadingState = consumeState<Loading>()
+ *   // By default, will drop any unmatched intermediate emissions
+ *   val loadedState = consumeState<Loaded>()
+ *   // Optionally, disallow dropping to expect the next emission to be the expected type
+ *   val loadedState = consumeState<Loaded>(dropUnmatchedIntermediates = false)
+ * }
+ * ```
+ */
+public suspend inline fun <reified SubState, UiState> CircuitReceiveTurbine<UiState>.consumeState(
+  dropUnmatchedIntermediates: Boolean = true
```

Comment:
I would be really wary of this in presenter tests.

We have already had a push internally to get rid of a lot of usage of `ReceiveTurbine.skipItem`, because it harms test readability and can make refactoring/altering these tests a pain. 

A `consumeState` with a conditional filter has the same problems, and also makes it easier to accidentally obscure faulty behavior in the presenter: the presenter may be emitting intermediate states that should be setting off alarms.

Reply:
Removed!